twoopstracker/authentication/forms.py

```diff
@@ -0,0 +1,61 @@
+from allauth.account import app_settings
+from allauth.account.adapter import get_adapter
+from allauth.account.forms import ResetPasswordForm as DefaultPasswordResetForm
+from allauth.account.forms import default_token_generator
+from allauth.account.utils import user_pk_to_url_str, user_username
+from django.conf import settings
+from django.contrib.sites.shortcuts import get_current_site
+from django.urls import reverse
+
+
+class CustomResetPasswordForm(DefaultPasswordResetForm):
+    def save(self, request, **kwargs):
+        email = self.cleaned_data["email"]
+        current_site = get_current_site(request)
+
+        if not self.users:
+            self._send_unknown_account_mail(request, email, current_site)
+        else:
+            self._send_password_reset_mail(
+                request, email, current_site, self.users, **kwargs
+            )
+        return
```

Comment:
Do we need to override this? both `_send_unknown_account_mail` and `_send_password_reset_mail` receive request so they should be able to get the `current_site` if they need to.

Reply:
@kilemensi I am overriding both of these not solely because of `current_site`,  but to update `signup_url` and `password_reset_url` on this functions. And yes, that current site can be down in both function (not passed as a parameter)

Here is what's happening: As we are using `dj_rest_auth` paths,.. in their [password reset serializers](https://github.com/iMerica/dj-rest-auth/blob/be436756793140621c6d6b1e82ba459676090b43/dj_rest_auth/serializers.py#L227), if allAuth is installed, they are using a custom allauth form which overrides the entire `save` function.

But looking at the [allauth form](https://github.com/pennersr/django-allauth/blob/ced1ddc730c36eca3551406c60e1577e30e01cbd/allauth/account/forms.py#L511), they have recent changes on their `save`, that dj_rest_auth custom form doesn't.

So here, I am overriding the save function..and including the new changes on allauth